node/lib/util/open.js

```diff
@@ -60,15 +60,18 @@ const SubmoduleFetcher    = require("./submodule_fetcher");
  * @param {String}           submoduleName
  * @param {String}           submoduleSha
  * @param {String|null}      templatePath
+ * @param {boolean}          bare
  * @return {NodeGit.Repository}
  */
 exports.openOnCommit = co.wrap(function *(fetcher,
                                           submoduleName,
                                           submoduleSha,
-                                          templatePath) {
+                                          templatePath, 
+                                          bare=false) {
```

Comment:
This doesn't do what you think it does. It assigns the local var "bare" to false (as well as passing false as the 5th arg to that function). Incidentally, I filed a jshint bug to ask them to flag this construct, and they rejected it because "I'm not sure why anyone would expect Python semantics from the given code example." Well, jshint folks, that's because Python is a language that people actually like and wish they were using, because it doesn't have nearly as many weird gotchas like this one.

Reply:
ack